src/dotnet/commands/dotnet-test/Program.cs

```diff
@@ -42,15 +42,35 @@ public int DoRun(string[] args)
                     RegisterForParentProcessExit(dotnetTestParams.ParentProcessId.Value);
                 }
 
-                var projectContexts = CreateProjectContexts(dotnetTestParams.ProjectPath);
-
-                var projectContext = projectContexts.First();
-
-                var testRunner = projectContext.ProjectFile.TestRunner;
+                var projectContexts = CreateProjectContexts(dotnetTestParams.ProjectPath, dotnetTestParams.Runtime);
+                var ranTests = false;
+                foreach (var projectContext in projectContexts)
+                {
+                    if (dotnetTestParams.Framework != null && dotnetTestParams.Framework != projectContext.TargetFramework)
+                    {
+                        continue;
+                    }
+
+                    ranTests = true;
+                    var testRunner = projectContext.ProjectFile.TestRunner;
+                    var dotnetTestRunner = _dotnetTestRunnerFactory.Create(dotnetTestParams.Port);
+                    var result = dotnetTestRunner.RunTests(projectContext, dotnetTestParams);
+                    if (result != 0)
+                    {
+                        return result;
```

Comment:
Suggestions on what the behavior should be here? Should we continue running tests for all tfms or fail as soon as one fails?